server/metadata.go

```diff
@@ -561,6 +603,7 @@ func (m *metadataAPI) addPartition(protoPartition *proto.Partition, recovered bo
 	if err != nil {
 		return nil, err
 	}
+	partition.paused = false
 	st.partitions[protoPartition.Id] = partition
```

Comment:
I'd have to check if there are other references, but we'll want to ensure that the old, paused partition that is replaced gets properly garbage collected.

Reply:
Alternatively, rather than using `CreatePartition` to unpause partitions, I wonder if it would make more sense to add a new Raft operation `UnpausePartition` that's able to resume an existing paused partition rather than creating a new one in its place? I could be convinced either way, just want to make sure we don't have any unforeseen memory leaks.

Reply:
I have used `CreatePartition` here because it seemed simpler to re-create the partition using the same data path, as you suggested. What would be the steps involved in resuming the partition without re-creating it? Would re-starting the commit log and calling `startLeadingOrFollowing` be enough?

Reply:
I think using `CreatePartition` is fine here.